api/app/hfi/hfi_admin.py

```diff
@@ -149,11 +149,19 @@ def get_next_order(updated_stations: List[PlanningWeatherStation], other_station
         Updated stations will include and removals, so that list may have a smaller max order.
     """
     updated_orders = [station.order_of_appearance_in_planning_area_list for station in updated_stations]
-    existing_orders = [station.order_of_appearance_in_planning_area_list for station in other_stations]
+
+    # An existing could be removed and hence have no order
+    existing_orders = [
+        station.order_of_appearance_in_planning_area_list for station in other_stations
+        if station.order_of_appearance_in_planning_area_list is not None]
 
     if len(updated_orders) == 0:
         return max(existing_orders) + 1
     if len(existing_orders) == 0:
         return max(updated_orders) + 1
 
     return min(max(updated_orders) + 1, max(existing_orders) + 1)
```

Comment:
I'd prefer to have a comment here explaining the logic behind this. I'm trying to wrap my head around it but it's a bit complicated and not immediately obvious what's going on here.

Reply:
I simplified the logic here after some further testing and updated the comment.